File: src/compute_client.cpp
#include <ros/ros.h>
#include "tsu200/Compute.h"
#include <cstdlib>

int main(int argc, char **argv) {
  ros::init(argc, argv, "compute_client_cpp");
  ros::NodeHandle nh;
  ros::ServiceClient client = nh.serviceClient<tsu200::Compute>("compute");
  tsu200::Compute srv;
  int a = 1, b = 2;
  if (argc > 2) {
    a = std::atoi(argv[1]);
    b = std::atoi(argv[2]);
  }
  srv.request.a = a;
  srv.request.b = b;
  if (client.call(srv)) {
    ROS_INFO("Result: %d", srv.response.sum);
  } else {
    ROS_ERROR("Failed to call service compute");
    return 1;
  }
  return 0;
}

File: scripts/compute_client.py
#!/usr/bin/env python
import sys
import rospy
from tsu200.srv import Compute


def compute_client(a, b):
    rospy.wait_for_service('compute')
    try:
        compute = rospy.ServiceProxy('compute', Compute)
        resp = compute(a, b)
        return resp.sum
    except rospy.ServiceException as e:
        rospy.logerr("Service call failed: %s", e)


if __name__ == "__main__":
    rospy.init_node('compute_client')
    a = int(sys.argv[1]) if len(sys.argv) > 1 else 1
    b = int(sys.argv[2]) if len(sys.argv) > 2 else 2
    result = compute_client(a, b)
    print("Result:", result)
